pygfx/objects/_base.py

```diff
@@ -389,153 +406,75 @@ def iter(self, filter_fn=None, skip_invisible=False):
         elif filter_fn(self):
             yield self
 
-        for child in self._children:
+        for child in self.children:
             yield from child.iter(filter_fn, skip_invisible)
 
-    def update_matrix(self):
-        p, r, s = self.position, self.rotation, self.scale
-        hash = p.x, p.y, p.z, r.x, r.y, r.z, r.w, s.x, s.y, s.z
-        if hash != self._transform_hash:
-            self._transform_hash = hash
-            self._matrix.compose(self.position, self.rotation, self.scale)
-            self._matrix_world_dirty = True
+    def get_bounding_box(self):
+        include_self = self.geometry is not None
+        n_partials = len(self.children) + 1 if include_self else len(self.children)
 
-    @property
-    def matrix(self):
-        """The (settable) transformation matrix."""
-        return self._matrix
-
-    @matrix.setter
-    def matrix(self, matrix):
-        self._matrix.copy(matrix)
-        self._matrix.decompose(self.position, self.rotation, self.scale)
-        self._matrix_world_dirty = True
+        if n_partials == 0:
+            # empty object with no mesh
+            return np.zeros((2, 3), dtype=float)
 
-    @property
-    def matrix_world(self):
-        """The world matrix (local matrix composed with any parent matrices)."""
-        return self._matrix_world
+        partial_aabb = np.zeros((n_partials, 2, 3), dtype=float)
+        for idx, child in enumerate(self.children):
+            aabb = child.get_bounding_box()
+            trafo = child.local.matrix
+            partial_aabb[idx] = la.aabb_transform(aabb, trafo)
+        if include_self:
+            partial_aabb[-1] = self.geometry.bounding_box()
 
-    @property
-    def matrix_auto_update(self):
-        """Whether or not the matrix auto-updates."""
-        return self._matrix_auto_update
+        final_aabb = np.zeros((2, 3), dtype=float)
+        final_aabb[0] = np.min(partial_aabb[:, 0, :], axis=0)
+        final_aabb[1] = np.max(partial_aabb[:, 1, :], axis=0)
```

Comment:
it always bothers me that there is no way to collect min and max in a single scan over the array :) do you happen to know of a method?

Reply:
I see, awesome! Thanks.